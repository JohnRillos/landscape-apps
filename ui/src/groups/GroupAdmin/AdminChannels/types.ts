import { Channel } from '@/types/groups';

export interface SectionMap {
  [key: string]: SectionListItem;
}

export interface ChannelListItem {
  key: string;
  channel: Channel;
}

export interface SectionListItem {
  title: string;
  isNew?: boolean;
  channels: ChannelListItem[];
}
