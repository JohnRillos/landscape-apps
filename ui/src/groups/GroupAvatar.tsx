import cn from 'classnames';
import React from 'react';
import ColorBoxIcon from '@/components/icons/ColorBoxIcon';
import { isColor } from '@/logic/utils';
import useMedia from '@/logic/useMedia';
import { useCalm } from '@/state/settings';

interface GroupAvatarProps {
  image?: string;
  size?: string;
  className?: string;
  title?: string;
}

const textSize = (size: string) => {
  const dims = parseInt(size.replace(/[^0-9.]/g, ''), 10);
  switch (dims) {
    case 7272:
      return 'text-3xl';
    case 2020:
      return 'text-xl';
    case 1616:
      return 'text-xl';
    case 1414:
      return 'text-xl';
    case 1212:
      return 'text-xl';
    case 66:
      return 'text-sm';
    default:
      return 'text-sm';
  }
};

export default function GroupAvatar({
  image,
  size = 'h-6 w-6',
  className,
  title,
}: GroupAvatarProps) {
  const dark = useMedia('(prefers-color-scheme: dark)');
  const calm = useCalm();
  let background;

  if (!calm.disableRemoteContent) {
    background = image || (dark ? '#333333' : '#E5E5E5');
  } else {
    background = dark ? '#333333' : '#E5E5E5';
  }

  return image && !calm.disableRemoteContent && !isColor(image) ? (
    <img className={cn('rounded', size, className)} src={image} />
  ) : (
    <ColorBoxIcon
      className={cn('rounded', size, textSize(size), className)}
      color={background}
      letter={title ? title.charAt(0) : ''}
    />
  );
}
