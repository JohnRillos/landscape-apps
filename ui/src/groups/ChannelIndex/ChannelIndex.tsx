/* eslint-disable no-console */
import { groupBy } from 'lodash';
import React, { useCallback, useState } from 'react';
import cn from 'classnames';
import { useNavigate } from 'react-router';
import { Link } from 'react-router-dom';
import { useRouteGroup, useGroup, useAmAdmin } from '@/state/groups';
import { Channel } from '@/types/groups';
import BubbleIcon from '@/components/icons/BubbleIcon';
import { channelHref, nestToFlag } from '@/logic/utils';
import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
import LeaveIcon from '@/components/icons/LeaveIcon';
import BulletIcon from '@/components/icons/BulletIcon';
import { useBriefs, useChatState } from '@/state/chat';
import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
import CaretDown16Icon from '@/components/icons/CaretDown16Icon';
import PencilIcon from '@/components/icons/PencilIcon';
import useRequestState from '@/logic/useRequestState';

const UNZONED = '';

function Channel({ channel, nest }: { nest: string; channel: Channel }) {
  const [app, flag] = nestToFlag(nest);
  const groupFlag = useRouteGroup();
  const group = useGroup(groupFlag);
  const briefs = useBriefs();
  const isAdmin = useAmAdmin(flag);
  const navigate = useNavigate();
  const [timer, setTimer] = useState<ReturnType<typeof setTimeout> | null>(
    null
  );
  const { isFailed, isPending, isReady, setFailed, setPending, setReady } =
    useRequestState();

  const editChannel = useCallback(() => {
    navigate(`/groups/${flag}/info/channels`);
  }, [flag, navigate]);

  const joinChannel = useCallback(async () => {
    try {
      if (timer) {
        clearTimeout(timer);
        setTimer(null);
      }
      setPending();
      await useChatState.getState().joinChat(flag);
      setReady();
    } catch (error) {
      if (error) {
        console.error(`[ChannelIndex:JoinError] ${error}`);
      }
      setFailed();
      setTimer(
        setTimeout(() => {
          setReady();
          setTimer(null);
        }, 10 * 1000)
      );
    }
  }, [flag, setFailed, setPending, setReady, timer]);

  const leaveChannel = useCallback(async () => {
    try {
      await useChatState.getState().leaveChat(flag);
    } catch (error) {
      if (error) {
        console.error(`[ChannelIndex:LeaveError] ${error}`);
      }
    }
  }, [flag]);

  const muteChannel = useCallback(() => {
    // TODO: what happens on Mute?
    console.log('mute ...');
  }, []);

  if (!group) {
    return null;
  }

  // If the current user is the Channel host, they are automatically joined,
  // and cannot leave the group
  const isChannelHost = window.our === flag?.split('/')[0];

  // A Channel is considered Joined if hosted by current user, or if a Brief
  // exists
  const joined = isChannelHost || (flag && flag in briefs);

  return (
    <div className="my-2 flex flex-row items-center justify-between rounded-lg pl-0 pr-2 hover:bg-gray-50">
      {/* avatar, title, participants */}
      <div className="flex flex-row">
        <div className="mr-3 flex h-12 w-12 items-center justify-center rounded bg-gray-50">
          {/* TODO: Channel Type icons */}
          <BubbleIcon className="h-6 w-6 text-gray-400" />
        </div>
        <div className="flex flex-col justify-evenly">
          {joined && nest ? (
            <Link
              className="font-semibold text-gray-800"
              to={channelHref(groupFlag, nest)}
            >
              {channel.meta.title}
            </Link>
          ) : (
            <div className="font-semibold text-gray-800">
              {channel.meta.title}
            </div>
          )}
        </div>
      </div>
      {/* action and options */}
      <div>
        {joined ? (
          <DropdownMenu.Root>
            <DropdownMenu.Trigger asChild className="appearance-none">
              <button className="button bg-green-soft text-green mix-blend-multiply dark:bg-green-900 dark:mix-blend-screen hover:dark:bg-green-800">
                <span className="mr-1">Joined</span>{' '}
                <CaretDown16Icon className="h-4 w-4" />
              </button>
            </DropdownMenu.Trigger>
            <DropdownMenu.Content className="dropdown">
              <DropdownMenu.Item
                onSelect={muteChannel}
                className="dropdown-item flex items-center space-x-2"
              >
                <BulletIcon className="h-6 w-6 text-gray-400" />
                <span className="text-gray-800">Mute Channel</span>
              </DropdownMenu.Item>
              {isAdmin ? (
                <DropdownMenu.Item
                  onSelect={editChannel}
                  className="dropdown-item flex items-center space-x-2"
                >
                  <PencilIcon className="m-1.5 h-3 w-3 fill-gray-500" />
                  <span>Edit Channel</span>
                </DropdownMenu.Item>
              ) : null}
              {!isChannelHost ? (
                <DropdownMenu.Item
                  onSelect={leaveChannel}
                  className="dropdown-item flex items-center space-x-2 text-red hover:bg-red-soft hover:dark:bg-red-900"
                >
                  <LeaveIcon className="h-6 w-6" />
                  <span>Leave Channel</span>
                </DropdownMenu.Item>
              ) : null}
            </DropdownMenu.Content>
          </DropdownMenu.Root>
        ) : (
          <button
            disabled={isPending}
            onClick={joinChannel}
            className={cn(
              'button mix-blend-multiply disabled:bg-gray-50 dark:mix-blend-screen',
              {
                'bg-gray-50': isReady || isPending,
                'bg-red-soft': isFailed,
                'text-gray-800': isReady,
                'text-gray-400': isPending,
                'text-red': isFailed,
              }
            )}
          >
            {isPending ? (
              <span className="center-items flex">
                <LoadingSpinner />
                <span className="ml-2">Joining...</span>
              </span>
            ) : isFailed ? (
              'Retry'
            ) : (
              'Join'
            )}
          </button>
        )}
      </div>
    </div>
  );
}

function ChannelSection({
  channels,
  zone,
}: {
  channels: [string, Channel][];
  zone: string | null;
}) {
  const sortedChannels = channels.slice();
  sortedChannels.sort(([, a], [, b]) =>
    a.meta.title.localeCompare(b.meta.title)
  );

  return (
    <div>
      {zone !== UNZONED ? (
        <div className="py-4 font-semibold text-gray-400">{zone}</div>
      ) : null}
      {sortedChannels.map(([nest, channel]) => (
        <Channel nest={nest} channel={channel} key={channel.added} />
      ))}
    </div>
  );
}

export default function ChannelIndex() {
  const flag = useRouteGroup();
  const group = useGroup(flag);
  const navigate = useNavigate();
  const isAdmin = useAmAdmin(flag);

  if (!group) {
    return null;
  }

  const sectionedChannels = groupBy(
    Object.entries(group.channels),
    ([, ch]) => ch.zone
  );
  // unsectioned channels have zone 'null' after groupBy; replace with empty str
  if ('null' in sectionedChannels) {
    sectionedChannels[UNZONED] = sectionedChannels.null;
    delete sectionedChannels.null;
  }

  const zones = Object.keys(sectionedChannels);
  // TODO: respect the sorted order set by the user?
  zones.sort((a, b) => {
    if (a === UNZONED) {
      return -1;
    }
    if (b === UNZONED) {
      return 1;
    }

    return a.localeCompare(b);
  });

  return (
    <section className="w-full p-4">
      <div className="mb-4 flex flex-row justify-between">
        <h1 className="text-lg font-bold">All Channels</h1>
        {isAdmin ? (
          <button
            onClick={() => navigate(`/groups/${flag}/info/channels`)}
            className="rounded-md bg-gray-800 py-1 px-2 text-[12px] font-semibold leading-4 text-white"
          >
            Channel Settings
          </button>
        ) : null}
      </div>
      {zones.map((zone) => (
        <div
          key={zone}
          className="mb-2 w-full rounded-xl bg-white py-1 pl-4 pr-2"
        >
          <ChannelSection channels={sectionedChannels[zone]} zone={zone} />
        </div>
      ))}
    </section>
  );
}
