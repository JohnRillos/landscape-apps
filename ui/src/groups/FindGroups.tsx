import React, { useCallback, useEffect, useState } from 'react';
import { useGroupState, usePendingGangs } from '@/state/groups';
import ShipSelector, { ShipOption } from '@/components/ShipSelector';
import { Gangs } from '@/types/groups';
import useRequestState from '@/logic/useRequestState';
import GroupJoinList from './GroupJoinList';

// TODO: how to handle the invite link case?

export default function FindGroups() {
  const [foundGangs, setFoundGangs] = useState<Gangs | null>(null);
  const pendingGangs = usePendingGangs();
  const [shipSelectorShips, setShipSelectorShips] = useState<ShipOption[]>([]);
  const selectedShip =
    shipSelectorShips.length > 0 ? shipSelectorShips[0] : null;
  const presentedShip = selectedShip
    ? selectedShip.label || selectedShip.value
    : '';
  const { isPending, isReady, setPending, setReady } = useRequestState();

  const searchGroups = useCallback(async () => {
    if (!selectedShip) {
      return;
    }
    setFoundGangs(null);
    setPending();
    const results = await useGroupState.getState().index(selectedShip.value);
    setFoundGangs(results);
    setReady();
  }, [selectedShip, setPending, setReady]);

  useEffect(() => {
    if (shipSelectorShips.length > 0) {
      searchGroups();
    }
  }, [searchGroups, shipSelectorShips]);

  return (
    <div className="flex grow bg-gray-50">
      <div className="w-full max-w-3xl p-4">
        <section className="card mb-4 space-y-8 p-8">
          <h1 className="text-lg font-bold">Find Groups</h1>
          <div>
            <label htmlFor="flag" className="mb-1.5 block font-semibold">
              Join Groups via Nickname or Urbit ID
            </label>
            <div className="flex flex-col space-y-2">
              <ShipSelector
                ships={shipSelectorShips}
                setShips={setShipSelectorShips}
                isMulti={false}
                isClearable={true}
                isLoading={isPending}
                hasPrompt={false}
                placeholder={''}
              />
            </div>
          </div>
          {selectedShip ? (
            <section className="card mb-4 space-y-8 p-8">
              <p className="font-semibold text-gray-400">
                {isPending ? (
                  <>
                    <span>Searching for groups hosted by&nbsp;</span>
                    <span className="text-gray-800">{presentedShip}</span>
                    <span>...</span>
                  </>
                ) : isReady ? (
                  <>
                    <span>Groups hosted by&nbsp;</span>
                    <span className="text-gray-800">{presentedShip}</span>
                    <span>:</span>
                  </>
                ) : null}
              </p>
              {isReady && foundGangs ? (
                <GroupJoinList gangs={foundGangs} />
              ) : isReady ? (
                <p>No groups found for &apos;{presentedShip}&apos;</p>
              ) : null}
            </section>
          ) : null}
        </section>
        {Object.keys(pendingGangs).length > 0 ? (
          <section className="card mb-4 space-y-8 p-8">
            <h1 className="text-lg font-bold">Pending Invites</h1>
            <GroupJoinList gangs={pendingGangs} />
          </section>
        ) : null}
      </div>
    </div>
  );
}
