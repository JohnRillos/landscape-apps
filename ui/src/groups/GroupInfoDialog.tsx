import React from 'react';
import GroupAvatar from './GroupAvatar';
import { useGroup, useRouteGroup } from '../state/groups';
import { useDismissNavigate } from '../logic/routing';
import Dialog, {
  DialogClose,
  DialogContent,
  DialogTitle,
} from '../components/Dialog';

export default function GroupInfoDialog() {
  const dismiss = useDismissNavigate();
  const flag = useRouteGroup();
  const group = useGroup(flag);
  const meta = group?.meta;

  const onOpenChange = (isOpen: boolean) => {
    if (!isOpen) {
      dismiss();
    }
  };

  if (!meta) {
    return null;
  }

  return (
    <Dialog defaultOpen onOpenChange={onOpenChange}>
      <DialogContent className="w-full sm:max-w-lg">
        <header className="flex items-center">
          <DialogTitle className="text-lg font-bold">Group Info</DialogTitle>
        </header>
        <div className="mt-6">
          <div className="flex flex-col items-center">
            <GroupAvatar img={meta.image} size="h-20 w-20" />
            <div className="my-4 text-center">
              <h2 className="center mb-2 font-semibold">{meta.title}</h2>
              {/* Current group meta object doesn't contain public/private info  */}
              <h3 className="text-base text-gray-600">Private Group</h3>
            </div>
            <p className="w-full leading-5">{meta.description}</p>
          </div>
        </div>
        <footer className="mt-8 flex items-center">
          <DialogClose asChild>
            <button className="button ml-auto">Close</button>
          </DialogClose>
        </footer>
      </DialogContent>
    </Dialog>
  );
}
