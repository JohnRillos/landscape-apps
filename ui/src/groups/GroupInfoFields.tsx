import React, { useEffect, useState } from 'react';
import { useFormContext } from 'react-hook-form';
import ColorPicker from '@/components/ColorPicker';
import ColorBoxIcon from '@/components/icons/ColorBoxIcon';
import EmptyIconBox from '@/components/icons/EmptyIconBox';
import useMedia from '@/logic/useMedia';
import { isValidUrl } from '@/logic/utils';
import { GroupMeta } from '@/types/groups';
import GroupAvatar from './GroupAvatar';

export default function GroupInfoFields() {
  const {
    register,
    unregister,
    watch,
    setValue,
    formState: { errors },
  } = useFormContext<GroupMeta>();
  const [iconType, setIconType] = useState<'image' | 'color'>();
  const [iconColor, setIconColor] = useState<string>();
  const [iconLetter, setIconLetter] = useState<string>();
  const [iconUrl, setIconUrl] = useState<string>();
  const watchIconColor = watch('color');
  const watchIconImage = watch('image');
  const watchTitle = watch('title');
  const dark = useMedia('(prefers-color-scheme: dark)');
  const defaultColor = dark ? '#FFFFFF' : '#000000';

  useEffect(() => {
    if (watchIconImage) {
      setIconType('image');
    } else if (watchIconColor) {
      setIconType('color');
    } else {
      setIconType(undefined);
    }
  }, [watchIconColor, watchIconImage]);

  useEffect(() => {
    if (iconType === 'color' && watchIconColor !== '') {
      setIconColor(watchIconColor as string);
    }
  }, [iconType, watchIconColor]);

  useEffect(() => {
    if (iconType === 'color' && watchTitle !== '') {
      setIconLetter((watchTitle as string).slice(0, 1));
    }
  }, [iconType, watchTitle]);

  useEffect(() => {
    if (iconType === 'image' && watchIconImage !== '') {
      setIconUrl(watchIconImage as string);
    }
  }, [iconType, watchIconImage]);

  useEffect(() => {
    register('color');
  }, []); // eslint-disable-line

  const handleCancelColorIcon = (e: React.SyntheticEvent) => {
    e.preventDefault();
    setIconType(undefined);
    setIconColor(undefined);
    setIconLetter(undefined);
    setValue('color', '');
  };

  const handleCancelImageIcon = (e: React.SyntheticEvent) => {
    e.preventDefault();
    setIconType(undefined);
    setIconUrl(undefined);
    setValue('image', '');
    unregister('image');
  };

  console.log(watchIconImage, iconUrl, isValidUrl(iconUrl));

  return (
    <>
      <div className="flex justify-between">
        <div className="flex flex-col">
          <span className="pb-2 font-bold">Group Icon *</span>
          <div className="flex items-center space-x-2">
            {iconType === undefined ? (
              <>
                <button
                  className="secondary-button"
                  onClick={() => setIconType('image')}
                >
                  Image URL
                </button>
                <span className="text-sm font-semibold">or</span>
                <button
                  className="secondary-button"
                  onClick={() => setIconType('color')}
                >
                  Fill Color
                </button>
              </>
            ) : null}
            {iconType === 'image' ? (
              <>
                <input
                  className="input"
                  placeholder="Paste Image URL"
                  {...register('image', {
                    required: true,
                    validate: (value) => isValidUrl(value),
                  })}
                  type="url"
                />
                <button
                  className="secondary-button"
                  onClick={handleCancelImageIcon}
                >
                  Cancel
                </button>
              </>
            ) : null}
            {iconType === 'color' ? (
              <div className="flex items-center space-x-2">
                <ColorPicker
                  color={
                    (watchIconColor as string) === ''
                      ? defaultColor
                      : (watchIconColor as string)
                  }
                  setColor={(newColor: string) =>
                    setValue('color', newColor || '', {
                      shouldDirty: true,
                      shouldTouch: true,
                    })
                  }
                />
                <button
                  className="secondary-button"
                  onClick={handleCancelColorIcon}
                >
                  Cancel
                </button>
              </div>
            ) : null}
            {errors.color ? (
              <span className="text-sm">{errors.color.message}</span>
            ) : null}
          </div>
        </div>
        {iconType === 'color' ? (
          <ColorBoxIcon
            className="h-12 w-12 text-xl"
            color={iconColor ? iconColor : '#000000'}
            letter={iconLetter ? iconLetter : 'T'}
          />
        ) : null}
        {iconType === 'image' && isValidUrl(iconUrl) ? (
          <GroupAvatar size="h-14 w-14" image={iconUrl} />
        ) : null}
        {iconType === undefined ? (
          <EmptyIconBox className="h-14 w-14 text-gray-300" />
        ) : null}
      </div>
      <div className="flex flex-col">
        <label htmlFor="title" className="pb-2 font-bold">
          Group Name *
        </label>
        <input
          // TODO: set sane maxLength
          {...register('title', { required: true, maxLength: 180 })}
          className="input"
          type="text"
          placeholder="Title"
        />
      </div>
      <div className="flex flex-col">
        <label htmlFor="description" className="pb-2 font-bold">
          Group Description (optional)
        </label>
        <textarea
          // TODO: set sane maxLength
          {...register('description', { maxLength: 300 })}
          className="input"
        />
      </div>
    </>
  );
}
