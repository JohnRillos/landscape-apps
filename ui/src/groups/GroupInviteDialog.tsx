import React, { useCallback } from 'react';
import Dialog, { DialogClose, DialogContent } from '../components/Dialog';
import { useDismissNavigate } from '../logic/routing';
import { useRouteGroup } from '../state/groups';

export default function GroupInviteDialog() {
  const dismiss = useDismissNavigate();
  const flag = useRouteGroup();

  const onOpenChange = (isOpen: boolean) => {
    if (!isOpen) {
      dismiss();
    }
  };

  const onInvite = useCallback(
    (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => null,
    []
  ); // TODO: invite poke

  return (
    <Dialog defaultOpen onOpenChange={onOpenChange}>
      <DialogContent containerClass="max-w-md" showClose>
        <div className="flex flex-col">
          <h2 className="mb-4 text-lg font-bold">Invite To Group</h2>
          <p className="mb-7 leading-5">Enter a username to invite to {flag}</p>
          <div className="flex items-center justify-end space-x-2">
            <DialogClose className="button" type="button">
              Cancel
            </DialogClose>

            <DialogClose
              onClick={onInvite}
              className="button bg-blue text-white"
            >
              Invite
            </DialogClose>
          </div>
        </div>
      </DialogContent>
    </Dialog>
  );
}
