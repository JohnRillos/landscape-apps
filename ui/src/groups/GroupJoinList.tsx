import React, { useCallback } from 'react';
import { useNavigate } from 'react-router';
import { useGang, useGroup, useGroupState } from '@/state/groups';
import GroupSummary, { GroupSummarySize } from './GroupSummary';

interface GroupJoinItemProps {
  flag: string;
  size?: GroupSummarySize;
}

function GroupJoinItem({ flag, size }: GroupJoinItemProps) {
  const navigate = useNavigate();
  const gang = useGang(flag);
  const group = useGroup(flag);

  const join = useCallback(() => {
    if (group) {
      return navigate(`/groups/${flag}`);
    }

    return useGroupState.getState().join(flag, true);
  }, [flag, group, navigate]);

  return (
    <li className="flex items-center p-2">
      <GroupSummary flag={flag} {...gang.preview} size={size} />
      <button
        className="button ml-auto bg-blue-soft text-blue dark:bg-blue-900"
        onClick={join}
      >
        {group ? 'Open' : 'Join'}
      </button>
    </li>
  );
}
interface GroupJoinListProps {
  gangs: string[];
  size?: GroupSummarySize;
}

export default function GroupJoinList({ gangs, size }: GroupJoinListProps) {
  return (
    <ul>
      {gangs.map((g) => (
        <GroupJoinItem key={g} flag={g} size={size} />
      ))}
    </ul>
  );
}
