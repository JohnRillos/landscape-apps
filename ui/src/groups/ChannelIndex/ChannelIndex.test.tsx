/* eslint-disable import/no-extraneous-dependencies */
import { createChannel, createMockGroup } from '@/mocks/groups';
import { Group } from '@/types/groups';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import { render } from '../../../test/utils';
import ChannelIndex from './ChannelIndex';

const fakeGroup: Group = createMockGroup('Fake Group');
fakeGroup.channels[`~zod/tlon`] = createChannel('Fake Channel');

vi.mock('@/state/groups', () => ({
  useGroup: () => fakeGroup,
  useRouteGroup: () => null,
  useAmAdmin: () => true,
}));

describe('ChannelIndex', () => {
  it('renders as expected', () => {
    const { asFragment } = render(<ChannelIndex />);
    expect(asFragment()).toMatchSnapshot();
  });
});
