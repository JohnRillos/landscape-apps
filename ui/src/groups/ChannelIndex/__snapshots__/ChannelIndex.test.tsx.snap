// Vitest Snapshot v1

exports[`ChannelIndex > renders as expected 1`] = `
<DocumentFragment>
  <section
    class="w-full p-4"
  >
    <div
      class="mb-4 flex flex-row justify-between"
    >
      <h1
        class="text-lg font-bold"
      >
        All Channels
      </h1>
      <button
        class="rounded-md bg-gray-800 py-1 px-2 text-[12px] font-semibold leading-4 text-white"
      >
        Channel Settings
      </button>
    </div>
    <div
      class="mb-2 w-full rounded-xl bg-white py-1 pl-4 pr-2"
    >
      <ul />
    </div>
  </section>
</DocumentFragment>
`;
